### ISSUE TITLE
Connecting to FQDN peers with SNI is broken 

### ISSUE THREAD

**Issue by felixbrucker** [Chia-Network/chia-blockchain#17923]
Currently when a FQDN is used to connect to a peer which requires SNI to correctly route the request to the host, no connection can be established. This is due to chia itself resolving the host and using the resulting ip to connect to it, breaking SNI.

**Comment by wjblanke** [comment]
3.9 aiohttp supports SNI so maybe it can be added

https://github.com/aio-libs/aiohttp/pull/7543

What feature are you trying to implement exactly felix?

**Comment by felixbrucker** [comment]
I wanted to connect to a full node behind cloudflare, but it would be the same for users trying to connect to one being nginx, given that they host multiple servers on the same port, for example 443. This however is currently broken and does not work, as the correct endpoint to connect to can not be mapped as that info is missing (bc it connects to the resolved ip instead of connecting to the hostname in the config, and letting the library handle resolving and properly connecting to it).

**Comment by felixbrucker** [comment]
I did some testing, i think there are two approaches to solve this:
- the normal way: pass in the host to `ws_connect`, for example: `ws_connect("wss://some-fqdn.com/ws")` instead of `ws_connect("wss://1.2.3.4/ws")`. In this case aiohttp already handles everything correctly and a connection can be established.
- the ugly way: continue to pass in the resolved ip as url, but also pass through the `server_name` argument and `Host` header set to the fqdn, for example: 
```py
ws_connect(
  "wss://1.2.3.4/ws",
  server_hostname=unresolved_peer_info.host,
  headers={
    "Host": unresolved_peer_info.host,
  },
)
```

Currently aiohttp can not set the server_hostname for `ws_connect`, see https://github.com/aio-libs/aiohttp/pull/7942 (has been on main for 6 months)

**Comment by wjblanke** [comment]
Is this just farmers trying to connect to a node or do you want node-node traffic working like this?

**Comment by felixbrucker** [comment]
I'd love to see it work for both, but primary is farmer to node

**Comment by wjblanke** [comment]
Looks like we need a new release of main aiohttp as well

**Comment by felixbrucker** [comment]
for the ugly way, yes

i'd much prefer the correct way

**Comment by wjblanke** [comment]
node - node gossip may be an issue. i think it only supports ip. the bigger issue is the current strategy for nodes is to resolve as early as possible, so this is opposite of that.

**Comment by felixbrucker** [comment]
Yeah resolving happens in chia-blockchain before connecting and it makes sense for network based configs, like trusted peers, but i'm not sure for which other reasons this might happen, or is needed even.

**Comment by wjblanke** [comment]
In the interim would running a proxy work to map the IP connections to a domain name? Its kludgy but should work

**Comment by felixbrucker** [comment]
Nope, because if you map ip to domain in a proxy you could just host on the ip itself in the first place, which is not desired and possible in my case.

**Comment by jack60612** [comment]
it is in our backlog as i mentioned in discord, not sure if this is an especially priority issue however.

**Comment by danieljperry** [comment]
@felixbrucker is this something you still need today?

**Comment by felixbrucker** [comment]
No, i gave up on this working a long time ago 😂

**Comment by danieljperry** [comment]
closing
